### PR TITLE
Data Tier correlation for searchable_snapshot mount location

### PR DESCRIPTION
**Pull request by stefnestor**
Per @andreidan wisdom:

> searchable snapshots are mounted directly on the corresponding data tiers (namely, fully mounted indices on the `hot`, `cold` or `frozen` tiers (depending on the ILM phase where the searchable_action is defined) while the partially mounted indices are mounted on the `frozen` tier).

This points out the prerequisite of the corresponding data tier for the searchable_snapshot phase.

### CODE REVIEW

**Review by jrodewig — COMMENTED** [review]
Thanks for this clarification @stefnestor.

In addition to your change, I added a sentence to make it clearer that the action requires data tiers. I also moved the last sentence to the first paragraph.

This LGTM, but I'll wait on @andreidan's review before merging.

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

File: docs/reference/ilm/actions/ilm-searchable-snapshot.asciidoc
Comment (andreidan) [inline docs/reference/ilm/actions/ilm-searchable-snapshot.asciidoc:12]: Could we expand this a bit and say something like:

```
The `searchable_snapshot` action requires a system that uses <<data-tiers,data tiers>> as it will mount the index directly on the corresponding tier using the <<tier-preference-allocation-filter, `index.routing.allocation.include._tier_preference`>> setting.
```
Maybe @jrodewig has a better suggestion here? (I don't particularly like "a system that uses ..." but not sure how else to phrase this).

**Review by andreidan — APPROVED** [review]
I've left a minor suggestion, thanks for working on this Stef!

File: docs/reference/ilm/actions/ilm-searchable-snapshot.asciidoc
Comment (jrodewig) [reply docs/reference/ilm/actions/ilm-searchable-snapshot.asciidoc]: Thanks @andreidan. I pushed some changes to incorporate the `index.routing.allocation.include._tier_preference` setting content.

Comment (andreidan) [reply docs/reference/ilm/actions/ilm-searchable-snapshot.asciidoc]: Thanks James

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by stefnestor** [comment]
@jrodewig from https://github.com/elastic/sdh-cloud/issues/27075#issuecomment-878345402 confusion, can we also back port this applicable info to 7.12, 7.11, & 7.10? I know we back ported to 7.x (thank you!), but if we still back port to the other version it'd help as well. 🙏🏼

**Comment by jrodewig** [comment]
Hi @stefnestor. Thanks for reaching out.

Per our [backport guidelines](https://github.com/elastic/elasticsearch/blob/master/docs/README.asciidoc#backporting-doc-fixes), we typically don't backport doc changes to out-of-maintenance versions. However, I'll work on getting these in as it seems to be a problem point.

**Pull request merged**
